lib/modules/sync_seeds.rb

```diff
@@ -0,0 +1,126 @@
+class SyncSeeds
+  require 'net/ssh'
+  require 'net/scp' 
+
+  def initialize(server, username)
+    @username = username
+    @server = server
+  end
+
+  def start_session
+    Net::SSH.start(@server, @username) do |session| 
+      @session = session
+      yield
+    end
+  end
+
+  def list_local_files(location)
+    Dir.glob('*', base: location)
+  end
+
+  def list_remote_files(location)
+    @session.sftp.dir.glob(location, '*')
+  end
+
+  def delete_files(files, location)
+    files.each do |file| 
+      puts "Removing #{file} as it no longer exists remotely"
+      FileUtils.rm_rf(File.join(location, file)) 
+    end
+  end
+
+  def files_for_deletion(list_of_files_1, list_of_files_2, location)
+    files = list_of_files_1 - list_of_files_2
+
+    if files.empty?
+      puts "No files to delete"
+    else
+      delete_files(files, location)
+    end
+  end
+
+  def download_files(source, dest)
+    puts "Downloading #{source} as it's newer or not found locally"
+    return @session.scp.download!(source, dest) if File.file?(dest)
+    @session.scp.download!(source, dest, recursive: true)
+  end
+  
+  def create_paths(relative_path)
+    { local_path: File.join(LOCAL, relative_path), remote_path: File.join(REMOTE, relative_path) }
+  end
+
+  def check_if_newer(parent_folder, local_item, remote_item, remote_path, local_path, base = LOCAL)
```

Comment:
I notcied the import taks always downloads the same files regardless if I just run it once before.
After debugging this a bit I noticed this goes to the `else` statement of the first condition, so saying that the resource doesn't exist. I wonder if it's because of the first parameter passed to this method on line 104 perhaps? @stanleypliu

Reply:
It's due to the encoding, I thought I'd managed to squash all the issues to do with filenames that have non-ASCII characters but evidently some still slip through the cracks. This is only affecting the files folder for obvious reasons. I'm going to fix this.